### DISCUSSION TITLE
Ok, so it mostly works... Tell me what you think...

### DISCUSSION THREAD

**Discussion by ExtremeElectronics** [ExtremeElectronics/RC2040#11] [Ideas]
Please tell me what you think ?
Any additions, 
Anything broken. (please add an issue) 

Absolutely no guarantee what you suggest will be added in. Or any time scales.
![Screenshot 2022-04-16 232731](https://user-images.githubusercontent.com/102665314/163693616-e6040e05-b927-4cc5-91bb-264e6441fc65.jpg)

**Comment by Jimbob-B** [comment]
Works nicely for me. Thank you ... been on the fence for an RC2014 when I saw the yellow ones and then saw this with thoughts of my pico sat in the drawer gathering dust.

My pico is sat in a Cytron Maker Pi Pico board and I had to make some amendments to pins but finally after 50 million builds it worked (turned out it did not like my PNY 8GB micro SD card at all). Not much of a programmer / github user but would you like the amendments I did? I was thinking of a global define for the Cytron board to pick up the pin assignment changes o not disrupt your setup.

**Reply by ExtremeElectronics** [reply]
Sorry I missed your post. 
Glad it worked for you.

**Comment by ExtremeElectronics** [comment]
Glad it worked for you. 

Thanks for the offer for the Cylon Board. Currently Id probably settle for better descriptions on how to change the ports. There are N+1 boards and possible setups out there and I'm not going to try to give options for them all. 

Also there is more stuff to come soon, so it will just give me more to back correct. :)

**Reply by Jimbob-B** [reply]
I sort of went from your code as some of the changes were commented out that I had to activate (SD card on SPI1, for example, for this particular board) and then moving the rom select buttons from GP10, GP11 and GP12 to other pins freed up from where your SD card was attached. Nothing major from memory ... it was just the SD card not working that got me confused!

Don't really know how to use CP/M but we'll see what happens!

**Comment by ExtremeElectronics** [comment]
OK, All looks good apart from the CPM on the SD, if you are getting different ROMs then the SD appears to work. 
The SD card needs every thing from here https://github.com/ExtremeElectronics/RC2040/tree/main/SD%20Card%20Contents 
(with the two files in the zip, unzipped) 

If I get a couple of mins Ill unzip the contents and try from the source myself to check the repository.

**Comment by ExtremeElectronics** [comment]
SD contents from 0.09 work fine with a new card on my Tiny2040 system
I have updated the utf2 (as it was the one I tested) from that release, but no one else has had an issue, and I know that has been used at least once for a new build since the week end. 

Sorry I'm out of ideas :(

**Comment by andypiper** [comment]
Arg, sorry for causing confusion / concern about the SD card and CP/M - user choice of terminal software error. I've now tried several other pieces of software (screen, picocom, dterm, and Serial.app) and there are no issues at all connecting! All working smoothly. I have attached a small speaker on GPIO 14/15 but I'm not getting any sound out from the sample apps - this could again easily be user error or misunderstanding, though.

I noticed the mario.bas lines wrap, and pasting that into MBASIC causes the second lines beginning `  20, ...` to overwrite the "real" line 20 in the program.

**Reply by ExtremeElectronics** [reply]
Glad you sorted it..

The music is the least tested of the code ASAIK only Ive tried it so far. You will need a PC speaker amp or a high impedance speaker/headphones to hear anything, the outputs of the Pico are only a couple of mA 

Regarding the program wrapping. When I paste the code into basic i don't get the wrapping. The lines come on the next line, but there isn't a cr/lf Does your terminal emulator add a cr/lf after 80 chars, or has the document editor added one before the paste?

**Reply by ExtremeElectronics** [reply]
Scrap that, its the upload to github that's wrapped the lines. I've updated the Mario.bas in examples.

**Comment by ExtremeElectronics** [comment]
New release 0.10-alpha, should fix sound issue. Would be useful to confirm if possible.
